hoomd/hpmc/tune/boxmc_move_size.py

```diff
@@ -0,0 +1,302 @@
+            }
+        params["max_move_size"] = max_move_size
+        self._param_dict.update(params)
+
+        self._update_tunables()
+
+    def attach(self, simulation):
+        if not isinstance(simulation.operations.integrator, HPMCIntegrator):
+            raise RuntimeError(
+                "MoveSizeTuner can only be used in HPMC simulations.")
+        super().attach(simulation)
+
+    def act(self, timestep=None):
+        """Tune move sizes.
+
+        Args:
+            timestep (`int`, optional): Current simulation timestep.
+        """
+        # update maximum move sizes
+        if self._should_update_tunables:
+            self._update_tunables()
+        if self._update_move_sizes:
+            for tunable in self._tunables:
+                max_move_size = self.max_move_size[tunable.attr]
+                tunable.domain = (self._min_move_size, max_move_size)
+        super().act(timestep)
+
+    def _update_tunables(self):
+        tunables = self._tunables
+
+        def filter_tunables(tunable):
+            return tunable.attr in self.moves
+
+        self._tunables = list(filter(filter_tunables, tunables))
+        tune_definitions = {t.attr for t in self._tunables}
+
+        # Add any move size tune definitions that are required by the new
+        # specification.
+        for move in self.moves:
+            if move in tune_definitions:
+                continue
+            max_move_size = self.max_move_size[move]
+            move_definition = _MoveSizeTuneDefinition(
+                self.boxmc, move, self.target,
+                (self._min_move_size, max_move_size))
+            self._tunables.append(move_definition)
+        self._should_update_tunables = False
+        self._tuned = 0
+
+    def _flag_move_size_update(self, value):
+        self._update_move_sizes = True
+        return value
+
+    def _flag_new_tunables(self, value):
+        self._should_update_tunables = True
+        return value
+        self._param_dict._cpp_obj = 5
+
+
+class BoxMCMoveSize(_InternalCustomTuner):
+    """Tunes ``BoxMC`` move sizes to targeted acceptance rate.
+
+    For most common creation of a `BoxMCMoveSize` tuner see
+    `BoxMCMoveSize.secant_solver` and `BoxMCMoveSize.scale_solver`
+    respectively.
+
+    Args:
+        trigger (hoomd.trigger.Trigger): ``Trigger`` to determine when to run
+            the tuner.
+        boxmc (hoomd.hpmc.update.BoxMC): The `hoomd.hpmc.update.BoxMC` object to
+            tune.
+        moves (list[str]): A list of types of moves to tune. Available options
+            are 'delta'.
+        target (float): The acceptance rate for trial moves that is desired. The
+            value should be between 0 and 1.
+        solver (`hoomd.tune.SolverStep`): A solver that tunes move sizes to
+            reach the specified target.
+        types (list[str]): A list of string particle types to tune the move
```

Comment:
Remove the types argument from the documentation. The class does not accept this argument.